### PR TITLE
[feature] output forwarding from bazel processes

### PR DESCRIPTION
**Pull request by lukaszwawrzyk**
Now, each bazel process run with `BazelRunner` (when `waitAndGetResult()` is called) will send log message to bsp-client:
Running: [command] and forward stdout and stderr. The output is also collected to string and lines list for later use.
PR also includes some refactoring to vavr, refactoring of bazel flags etc. 

I also replaced includedTargets and excludedTargets pair with single class TargetSpecs. In the future I imagine it as the main way to specify what should be built/run. It can also contain directories to derive targets from. Bazel runner could resolve what to run and run it.

### CODE REVIEW

**Review by abrams27 — COMMENTED** [review]
looks good, but the test fails - I couldn't spot the reason

**Comment by abrams27** [comment]
ohh and I thing that it would be nice to color (or make it bold if it is possible) `invoking: ...` line as well

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcess.java [hunk 1]
```diff
@@ -1,61 +1,34 @@
...
     this.process = process;
-    this.buildClientLogger = buildClientLogger;
+    this.logger = logger;
   }
 
   public BazelProcessResult waitAndGetResult() {
     try {
+      var stdoutCollector = new OutputCollector();
+      var stderrCollector = new OutputCollector();
+      AsyncOutputProcessor.run(process.getInputStream(), stdoutCollector, logger::logMessage);
```
Comment (abrams27) [inline bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcess.java:22]: we can add logging to file here as well

**Comment by lukaszwawrzyk** [comment]
> looks good, but the test fails - I couldn't spot the reason

I think I have a race condition. Working on a fix.

**Comment by lukaszwawrzyk** [comment]
> ohh and I thing that it would be nice to color (or make it bold if it is possible) `invoking: ...` line as well

The problem is that the sync console displays color but the build console doesn't (and you see some unprintable characters). So I just carefully enabled colors only on calls when it is useful and displays in sync console. If I colored the Invoking it would not always display correctly :/

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcess.java (hunk 1 above)
Comment (abrams27) [reply bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcess.java]: yep

**Review by abrams27 — APPROVED** [review]
please update changelog + logging to file

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +21,9 @@
 
 ### Changes 🔄
 
+- Improved bazel runner. The BSP Client now receives invoked bazel command,
```
Comment (abrams27) [inline CHANGELOG.md:24]: I think that we can consider it as a feature

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/utils/Stopwatch.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package org.jetbrains.bsp.bazel.bazelrunner.utils;
+
+import java.time.Duration;
+
+public class Stopwatch {
```
Comment (abrams27) [inline bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/utils/Stopwatch.java:5]: u can use guava stop watch as well if u want to: https://guava.dev/releases/24.0-jre/api/docs/com/google/common/base/Stopwatch.html

**Pull request merged**